# Profitable App Profiles for the App Store and Google Play Markets

Our aim in this project is to find mobile app profiles that are profitable for the App Store and Google Play markets. We're working as data analysts for a company that builds Android and iOS mobile apps, and our job is to enable our team of developers to make data-driven decisions with respect to the kind of apps they build.

At our company, we only build apps that are free to download and install, and our main source of revenue consists of in-app ads. This means that our revenue for any given app is mostly influenced by the number of users that use our app. Our goal for this project is to analyze data to help our developers understand what kinds of apps are likely to attract more users.

## Opening and Exploring the Data 

As of September 2018, there were approximately 2 million iOS apps available on the App Store, and 2.1 million Android apps on Google Play.

![img](https://s3.amazonaws.com/dq-content/350/py1m8_statista.png) Source: [Statista](https://www.statista.com/statistics/276623/number-of-apps-available-in-leading-app-stores/)
Collecting data for over four million apps requires a significant amount of time and money, so we'll try to analyze a sample of data instead. To avoid spending resources with collecting new data ourselves, we should first try to see whether we can find any relevant existing data at no cost. Luckily, these are two data sets that seem suitable for our purpose:

##### A data set containing data about approximately ten thousand Android apps from Google Play.
###### A data set containing data about approximately seven thousand iOS apps from the App Store. 
Let's start by opening the two data sets and then continue with exploring the data.

In [1]:
import pandas as pd
import numpy as np

In [2]:
android = pd.read_csv('googleplaystore.csv')
android.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [3]:
ios = pd.read_csv('Applestore.csv')
ios.head()

Unnamed: 0         id                                         track_name  \
0           1  281656475                                    PAC-MAN Premium   
1           2  281796108                          Evernote - stay organized   
2           3  281940292    WeatherBug - Local Weather, Radar, Maps, Alerts   
3           4  282614216  eBay: Best App to Buy, Sell, Save! Online Shop...   
4           5  282935706                                              Bible   

   size_bytes currency  price  rating_count_tot  rating_count_ver  \
0   100788224      USD   3.99             21292                26   
1   158578688      USD   0.00            161065                26   
2   100524032      USD   0.00            188583              2822   
3   128512000      USD   0.00            262241               649   
4    92774400      USD   0.00            985920              5320   

   user_rating  user_rating_ver     ver cont_rating   prime_genre  \
0          4.0              4.5   6.3.5          4+         Games   
1          4.0              3.5   8.2.2          4+  Productivity   
2          3.5              4.5   5.0.0          4+       Weather   
3          4.0              4.5  5.10.0         12+      Shopping   
4          4.5              5.0   7.5.1          4+     Reference   

   sup_devices.num  ipadSc_urls.num  lang.num  vpp_lic  
0               38                5        10        1  
1               37                5        23        1  
2               37                5         3        1  
3               37                5         9        1  
4               37                5        45        1

In [4]:
ios = ios.drop(['Unnamed: 0'], axis = 1)

In [5]:
ios.head()

id                                         track_name  size_bytes  \
0  281656475                                    PAC-MAN Premium   100788224   
1  281796108                          Evernote - stay organized   158578688   
2  281940292    WeatherBug - Local Weather, Radar, Maps, Alerts   100524032   
3  282614216  eBay: Best App to Buy, Sell, Save! Online Shop...   128512000   
4  282935706                                              Bible    92774400   

  currency  price  rating_count_tot  rating_count_ver  user_rating  \
0      USD   3.99             21292                26          4.0   
1      USD   0.00            161065                26          4.0   
2      USD   0.00            188583              2822          3.5   
3      USD   0.00            262241               649          4.0   
4      USD   0.00            985920              5320          4.5   

   user_rating_ver     ver cont_rating   prime_genre  sup_devices.num  \
0              4.5   6.3.5          4+         Games               38   
1              3.5   8.2.2          4+  Productivity               37   
2              4.5   5.0.0          4+       Weather               37   
3              4.5  5.10.0         12+      Shopping               37   
4              5.0   7.5.1          4+     Reference               37   

   ipadSc_urls.num  lang.num  vpp_lic  
0                5        10        1  
1                5        23        1  
2                5         3        1  
3                5         9        1  
4                5        45        1

In [8]:
android.nunique()

App               9660
Category            34
Rating              40
Reviews           6002
Size               462
Installs            22
Type                 3
Price               93
Content Rating       6
Genres             120
Last Updated      1378
Current Ver       2832
Android Ver         33
dtype: int64

In [9]:
ios.nunique()

id                  7197
track_name          7195
size_bytes          7107
currency               1
price                 36
rating_count_tot    3185
rating_count_ver    1138
user_rating           10
user_rating_ver       10
ver                 1590
cont_rating            4
prime_genre           23
sup_devices.num       20
ipadSc_urls.num        6
lang.num              57
vpp_lic                2
dtype: int64

## Deleting Wrong Data

The Google Play data set has a dedicated discussion section https://www.kaggle.com/lava18/google-play-store-apps/discussion, and we can see that one of the discussions outlines an error for row 10472. Let's print this row and compare it against the header and another row that is correct.

In [10]:
android.iloc[[10472]]

App Category  Rating Reviews  \
10472  Life Made WI-Fi Touchscreen Photo Frame      1.9    19.0    3.0M   

         Size Installs Type     Price Content Rating             Genres  \
10472  1,000+     Free    0  Everyone            NaN  February 11, 2018   

      Last Updated Current Ver Android Ver  
10472       1.0.19  4.0 and up         NaN

The row 10472 corresponds to the app Life Made WI-Fi Touchscreen Photo Frame, and we can see that the rating is 19. This is clearly off because the maximum rating for a Google Play app is 5 (as mentioned in the discussions section,https://www.kaggle.com/lava18/google-play-store-apps/discussion/66015 this problem is caused by a missing value in the 'Category' column). As a consequence, we'll delete this row.

In [11]:
android = android.drop(10472)

In [12]:
android.iloc[[10472]]

App Category  Rating Reviews  Size     Installs  \
10473  osmino Wi-Fi: free WiFi    TOOLS     4.2  134203  4.1M  10,000,000+   

       Type Price Content Rating Genres    Last Updated Current Ver  \
10473  Free     0       Everyone  Tools  August 7, 2018     6.06.14   

      Android Ver  
10473  4.4 and up

In [13]:
android.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

## Remove duplicate enteries 

### Part One
If we explore the Google Play data set long enough, we'll find that some apps have more than one entry. For instance, the application Instagram has four entries:

In [14]:
# Select duplicate rows except first occurrence based on all columns
duplicateRowsDF = android[android.duplicated()]
print("Duplicate Rows except first occurrence based on all columns are :")
print(duplicateRowsDF)

Duplicate Rows except first occurrence based on all columns are :
                                                   App      Category  Rating  \
229                       Quick PDF Scanner + OCR FREE      BUSINESS     4.2   
236                                                Box      BUSINESS     4.2   
239                                 Google My Business      BUSINESS     4.4   
256                                ZOOM Cloud Meetings      BUSINESS     4.4   
261                          join.me - Simple Meetings      BUSINESS     4.0   
...                                                ...           ...     ...   
8643                    Wunderlist: To-Do List & Tasks  PRODUCTIVITY     4.6   
8654   TickTick: To Do List with Reminder, Day Planner  PRODUCTIVITY     4.6   
8658                           ColorNote Notepad Notes  PRODUCTIVITY     4.6   
10049        Airway Ex - Intubate. Anesthetize. Train.       MEDICAL     4.3   
10768                                             AAFP

In total, there are 1,181 cases where an app occurs more than once and we have 483 rows that are duplicates:

In [ ]:
#length1 = len(android)

# sorting by first name
#android.sort_values("App", inplace=True)
 
# dropping duplicate values
#android.drop_duplicates(keep=False,inplace=True)
 
# length after removing duplicates
#length2=len(android)
 
# printing all data frame lengths
#print(length1, length2)

Examples of duplicate apps: ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']
We don't want to count certain apps more than once when we analyze data, so we need to remove the duplicate entries and keep only one entry per app. One thing we could do is remove the duplicate rows randomly, but we could probably find a better way.

If you examine the rows we printed two cells above for the Instagram app, the main difference happens on the fourth position of each row, which corresponds to the number of reviews. The different numbers show that the data was collected at different times. We can use this to build a criterion for keeping rows. We won't remove rows randomly, but rather we'll keep the rows that have the highest number of reviews because the higher the number of reviews, the more reliable the ratings.

In [17]:
android = android.sort_values('Reviews', ascending=False).drop_duplicates('App').sort_index()

In [18]:
len(android)

9659

# Removing Non-English Apps

In [59]:
print(ios.iloc[814][1])
print(ios.iloc[6734][1])

print(android.iloc[441][0])
print(android.iloc[7950][0])

搜狐新闻—新闻热点资讯掌上阅读软件
エレメンタル ファンタジー - 高精細３ＤアクションＲＰＧ
American Girls Mobile Numbers
Tuberculose Dz


We're not interested in keeping these kind of apps, so we'll remove them. One way to go about this is to remove each app whose name contains a symbol that is not commonly used in English text — English text usually includes letters from the English alphabet, numbers composed of digits from 0 to 9, punctuation marks (., !, ?, ;, etc.), and other symbols (+, *, /, etc.).

All these characters that are specific to English texts are encoded using the ASCII standard. Each ASCII character has a corresponding number between 0 and 127 associated with it, and we can take advantage of that to build a function that checks an app name and tells us whether it contains non-ASCII characters.

In [66]:
android = android[android.App.map(lambda x: x.isascii())]

In [64]:
ios = ios[ios.track_name.map(lambda x: x.isascii())]

In [65]:
print(ios.iloc[814][1])

Secret Calculator Pro - Hide photos & videos


In [67]:
print(len(android))
print(len(ios))

9117
5707


# Isolating the Free Apps
As we mentioned in the introduction, we only build apps that are free to download and install, and our main source of revenue consists of in-app ads. Our data sets contain both free and non-free apps, and we'll need to isolate only the free apps for our analysis. Below, we isolate the free apps for both our data sets.

In [74]:
android_final = android.loc[android['Price'] == '0']

In [76]:
android_final.head()

App        Category  Rating  \
0  Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
3                           Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4           Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   
5                      Paper flowers instructions  ART_AND_DESIGN     4.4   
6         Smoke Effect Photo Maker - Smoke Editor  ART_AND_DESIGN     3.8   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   
5     167  5.6M      50,000+  Free     0       Everyone   
6     178   19M      50,000+  Free     0       Everyone   

                    Genres     Last Updated         Current Ver   Android Ver  
0             Art & Design  January 7, 2018               1.0.0  4.0.3 and up  
3             Art & Design     June 8, 2018  Varies with device    4.2 and up  
4  Art & Design;Creativity    June 20, 2018                 1.1    4.4 and up  
5             Art & Design   March 26, 2017                 1.0    2.3 and up  
6             Art & Design   April 26, 2018                 1.1  4.0.3 and up

In [77]:
len(android_final)

8406

In [81]:
ios_final = ios.loc[ios['price'] == 0.0]

In [82]:
ios_final.head()

id                                         track_name  size_bytes  \
1  281796108                          Evernote - stay organized   158578688   
2  281940292    WeatherBug - Local Weather, Radar, Maps, Alerts   100524032   
3  282614216  eBay: Best App to Buy, Sell, Save! Online Shop...   128512000   
4  282935706                                              Bible    92774400   
6  283646709             PayPal - Send and request money safely   227795968   

  currency  price  rating_count_tot  rating_count_ver  user_rating  \
1      USD    0.0            161065                26          4.0   
2      USD    0.0            188583              2822          3.5   
3      USD    0.0            262241               649          4.0   
4      USD    0.0            985920              5320          4.5   
6      USD    0.0            119487               879          4.0   

   user_rating_ver     ver cont_rating   prime_genre  sup_devices.num  \
1              3.5   8.2.2          4+  Productivity               37   
2              4.5   5.0.0          4+       Weather               37   
3              4.5  5.10.0         12+      Shopping               37   
4              5.0   7.5.1          4+     Reference               37   
6              4.5  6.12.0          4+       Finance               37   

   ipadSc_urls.num  lang.num  vpp_lic  
1                5        23        1  
2                5         3        1  
3                5         9        1  
4                5        45        1  
6                0        19        1

In [83]:
len(ios_final)

2922

We're left with 8406 Android apps and 2922 iOS apps, which should be enough for our analysis.

# Most Common Apps by Genre
### Part One
As we mentioned in the introduction, our aim is to determine the kinds of apps that are likely to attract more users because our revenue is highly influenced by the number of people using our apps.

To minimize risks and overhead, our validation strategy for an app idea is comprised of three steps:

1. Build a minimal Android version of the app, and add it to Google Play.
2. If the app has a good response from users, we then develop it further.
3. If the app is profitable after six months, we also build an iOS version of the app and add it to the App Store.
Because our end goal is to add the app on both the App Store and Google Play, we need to find app profiles that are successful on both markets. For instance, a profile that might work well for both markets might be a productivity app that makes use of gamification.

Let's begin the analysis by getting a sense of the most common genres for each market. 

In [101]:
tab = pd.crosstab(index=ios_final['prime_genre'], columns='count')
#freq = (tab / tab.sum()) * 100
tab

col_0              count
prime_genre             
Book                   8
Business              15
Catalogs               3
Education            112
Entertainment        220
Finance               32
Food & Drink          26
Games               1729
Health & Fitness      58
Lifestyle             43
Medical                6
Music                 63
Navigation             4
News                  39
Photo & Video        150
Productivity          50
Reference             15
Shopping              73
Social Networking     91
Sports                60
Travel                33
Utilities             66
Weather               26

In [87]:
freq.sort_values('count', ascending=False)

col_0                  count
prime_genre                 
Games              59.171800
Entertainment       7.529090
Photo & Video       5.133470
Education           3.832991
Social Networking   3.114305
Shopping            2.498289
Utilities           2.258727
Music               2.156057
Sports              2.053388
Health & Fitness    1.984942
Productivity        1.711157
Lifestyle           1.471595
News                1.334702
Travel              1.129363
Finance             1.095140
Weather             0.889802
Food & Drink        0.889802
Reference           0.513347
Business            0.513347
Book                0.273785
Medical             0.205339
Navigation          0.136893
Catalogs            0.102669

We can see that among the free English apps, more than a half (59.17%) are games. Entertainment apps are close to 8%, followed by photo and video apps, which are approximately 5%. Only 3.83% of the apps are designed for education, followed by social networking apps which amount for 3.11% of the apps in our data set.

The general impression is that App Store (at least the part containing free English apps) is dominated by apps that are designed for fun (games, entertainment, photo and video, social networking, sports, music, etc.), while apps with practical purposes (education, shopping, utilities, productivity, lifestyle, etc.) are more rare. However, the fact that fun apps are the most numerous doesn't also imply that they also have the greatest number of users — the demand might not be the same as the offer.

Let's continue by examining the Genres and Category columns of the Google Play data set (two columns which seem to be related).

In [89]:
tab1 = pd.crosstab(index=android_final['Category'], columns='count')
freq1 = (tab1 / tab1.sum()) * 100

In [90]:
freq1.sort_values('count', ascending=False)

col_0                    count
Category                      
FAMILY               18.843683
GAME                  9.576493
TOOLS                 8.565310
BUSINESS              4.710921
PRODUCTIVITY          3.973352
LIFESTYLE             3.890079
FINANCE               3.735427
MEDICAL               3.628361
PERSONALIZATION       3.307162
SPORTS                3.259576
COMMUNICATION         3.223888
HEALTH_AND_FITNESS    3.128718
PHOTOGRAPHY           3.009755
NEWS_AND_MAGAZINES    2.795622
SOCIAL                2.664763
TRAVEL_AND_LOCAL      2.307875
SHOPPING              2.248394
BOOKS_AND_REFERENCE   2.188913
DATING                1.832025
VIDEO_PLAYERS         1.760647
MAPS_AND_NAVIGATION   1.356174
FOOD_AND_DRINK        1.201523
EDUCATION             1.177730
AUTO_AND_VEHICLES     0.939805
ENTERTAINMENT         0.927909
LIBRARIES_AND_DEMO    0.904116
HOUSE_AND_HOME        0.808946
WEATHER               0.797050
EVENTS                0.713776
ART_AND_DESIGN        0.666191
PARENTING             0.654295
BEAUTY                0.630502
COMICS                0.571021

The landscape seems significantly different on Google Play: there are not that many apps designed for fun, and it seems that a good number of apps are designed for practical purposes (family, tools, business, lifestyle, productivity, etc.). However, if we investigate this further, we can see that the family category (which accounts for almost 19% of the apps) means mostly games for kids.

https://play.google.com/store/apps/category/FAMILY?hl=en

Even so, practical apps seem to have a better representation on Google Play compared to App Store. This picture is also confirmed by the frequency table we see for the Genres column:

In [92]:
tab2 = pd.crosstab(index=android_final['Genres'], columns='count')
freq2 = (tab2 / tab2.sum()) * 100

In [93]:
freq2.sort_values('count', ascending=False)

col_0                                   count
Genres                                       
Tools                                8.553414
Entertainment                        6.090887
Education                            5.389008
Business                             4.710921
Productivity                         3.973352
...                                       ...
Art & Design;Pretend Play            0.011896
Puzzle;Education                     0.011896
Health & Fitness;Action & Adventure  0.011896
Health & Fitness;Education           0.011896
Lifestyle;Pretend Play               0.011896

[109 rows x 1 columns]

The difference between the Genres and the Category columns is not crystal clear, but one thing we can notice is that the Genres column is much more granular (it has more categories). We're only looking for the bigger picture at the moment, so we'll only work with the Category column moving forward.

Up to this point, we found that the App Store is dominated by apps designed for fun, while Google Play shows a more balanced landscape of both practical and for-fun apps. Now we'd like to get an idea about the kind of apps that have most users.

# Most Popular Apps by Genre on the App Store
One way to find out what genres are the most popular (have the most users) is to calculate the average number of installs for each app genre. For the Google Play data set, we can find this information in the Installs column, but for the App Store data set this information is missing. As a workaround, we'll take the total number of user ratings as a proxy, which we can find in the rating_count_tot app.

Below, we calculate the average number of user ratings per app genre on the App Store:

In [108]:
total = ios_final.groupby('prime_genre')['rating_count_tot'].sum()

prime_genre
Book                   133368
Business               102594
Catalogs                15585
Education              683588
Entertainment         3301370
Finance                833238
Food & Drink           866682
Games                37278538
Health & Fitness      1126280
Lifestyle              742203
Medical                  3672
Music                 3489949
Navigation             500149
News                   911905
Photo & Video         4387465
Productivity          1142111
Reference             1343439
Shopping              2108063
Social Networking     7149625
Sports                1547500
Travel                1125814
Utilities              763732
Weather               1255165
Name: rating_count_tot, dtype: int64

In [145]:
tab.index

Index(['Book', 'Business', 'Catalogs', 'Education', 'Entertainment', 'Finance',
       'Food & Drink', 'Games', 'Health & Fitness', 'Lifestyle', 'Medical',
       'Music', 'Navigation', 'News', 'Photo & Video', 'Productivity',
       'Reference', 'Shopping', 'Social Networking', 'Sports', 'Travel',
       'Utilities', 'Weather'],
      dtype='object', name='prime_genre')

In [161]:
total = total.to_frame()

In [172]:
dfd = total.div(tab.values)
dfd.columns = total.columns + '/' + tab.columns

In [176]:
dfd.sort_values('rating_count_tot/count',ascending =False)

rating_count_tot/count
prime_genre                              
Navigation                  125037.250000
Reference                    89562.600000
Social Networking            78567.307692
Music                        55396.015873
Weather                      48275.576923
Travel                       34115.575758
Food & Drink                 33333.923077
Photo & Video                29249.766667
Shopping                     28877.575342
Finance                      26038.687500
Sports                       25791.666667
News                         23382.179487
Productivity                 22842.220000
Games                        21560.750723
Health & Fitness             19418.620690
Lifestyle                    17260.534884
Book                         16671.000000
Entertainment                15006.227273
Utilities                    11571.696970
Business                      6839.600000
Education                     6103.464286
Catalogs                      5195.000000
Medical                        612.000000

On average, navigation apps have the highest number of user reviews, but this figure is heavily influenced by Waze and Google Maps, which have close to half a million user reviews together:

In [179]:
ios_final.loc[ios_final['prime_genre'] == 'Navigation']

id                                        track_name  size_bytes  \
174   323229106   Waze - GPS Navigation, Maps & Real-time Traffic    94139392   
280   344176018  ImmobilienScout24: Real Estate Search in Germany   126867456   
959   463431091                              Railway Route Search    46950400   
1693  585027354                Google Maps - Navigation & Transit   120232960   

     currency  price  rating_count_tot  rating_count_ver  user_rating  \
174       USD    0.0            345046              3040          4.5   
280       USD    0.0               187                 0          3.5   
959       USD    0.0                 5                 0          3.0   
1693      USD    0.0            154911              1253          4.5   

      user_rating_ver     ver cont_rating prime_genre  sup_devices.num  \
174               4.5    4.24          4+  Navigation               37   
280               0.0     9.5          4+  Navigation               37   
959               0.0  3.17.1          4+  Navigation               37   
1693              4.0  4.31.1         12+  Navigation               37   

      ipadSc_urls.num  lang.num  vpp_lic  
174                 5        36        1  
280                 5         3        1  
959                 0         1        1  
1693                5        34        1

The same pattern applies to social networking apps, where the average number is heavily influenced by a few giants like Facebook, Pinterest, Skype, etc. Same applies to music apps, where a few big players like Pandora, Spotify, and Shazam heavily influence the average number.

Our aim is to find popular genres, but navigation, social networking or music apps might seem more popular than they really are. The average number of ratings seem to be skewed by very few apps which have hundreds of thousands of user ratings, while the other apps may struggle to get past the 10,000 threshold. We could get a better picture by removing these extremely popular apps for each genre and then rework the averages, but we'll leave this level of detail for later.

Reference apps have 89,562 user ratings on average, but it's actually the Bible and Dictionary.com which skew up the average rating:

In [180]:
ios_final.loc[ios_final['prime_genre'] == 'Reference']

id                                         track_name  \
4      282935706                                              Bible   
116    308750436              Dictionary.com Dictionary & Thesaurus   
375    364740856     Dictionary.com Dictionary & Thesaurus for iPad   
503    388389451  Muslim Pro: Ramadan 2017 Prayer Times, Azan, Q...   
587    399452287                         Merriam-Webster Dictionary   
681    414706506                                   Google Translate   
1023   475772902                                          Night Sky   
1934   640199958                                               WWDC   
2084   671889349  Jishokun-Japanese English Dictionary & Translator   
3892  1003837100                                        VPN Express   
6462  1130829481  New Furniture Mods - Pocket Wiki & Game Tools ...   
6524  1133706938  Horror Maps for Minecraft PE - Download The Sc...   
6567  1135575003  City Maps for Minecraft PE - The Best Maps for...   
6613  1137683736    GUNS MODS for Minecraft PC Edition - Mods Tools   
6905  1156856246    Real Bike Traffic Rider Virtual Reality Glasses   

      size_bytes currency  price  rating_count_tot  rating_count_ver  \
4       92774400      USD    0.0            985920              5320   
116    111275008      USD    0.0            200047               177   
375    165748736      USD    0.0             54175             10176   
503    100551680      USD    0.0             18418               706   
587    155593728      USD    0.0             16849              1125   
681     65281024      USD    0.0             26786                27   
1023   596499456      USD    0.0             12122                60   
1934    34959360      USD    0.0               762                 7   
2084   225522688      USD    0.0                 0                 0   
3892    10645504      USD    0.0                14                 1   
6462    52959232      USD    0.0             17588             17588   
6524    89927680      USD    0.0               718               718   
6567    90124288      USD    0.0              8535              8535   
6613    85424128      USD    0.0              1497               245   
6905   125990912      USD    0.0                 8                 2   

      user_rating  user_rating_ver      ver cont_rating prime_genre  \
4             4.5              5.0    7.5.1          4+   Reference   
116           4.0              4.0    7.1.3          4+   Reference   
375           4.5              4.5      4.0          4+   Reference   
503           4.5              5.0    9.2.1          4+   Reference   
587           4.5              4.5      4.1          4+   Reference   
681           3.5              4.5   5.10.0          4+   Reference   
1023          4.5              4.5    4.4.1          4+   Reference   
1934          3.5              4.5    6.0.0          4+   Reference   
2084          0.0              0.0  2.10.14         12+   Reference   
3892          4.5              5.0      1.8          4+   Reference   
6462          4.5              4.5      1.0          4+   Reference   
6524          4.0              4.0      1.0         12+   Reference   
6567          4.0              4.0      1.0          4+   Reference   
6613          4.0              3.0      1.1         12+   Reference   
6905          3.0              4.5      8.9         17+   Reference   

      sup_devices.num  ipadSc_urls.num  lang.num  vpp_lic  
4                  37                5        45        1  
116                37                0         1        1  
375                24                5         9        1  
503                37                5        16        1  
587                38                1        12        1  
681                37                5        59        1  
1023               37                5        29        1  
1934               37                5         1        1  
2084               37                4         

However, this niche seems to show some potential. One thing we could do is take another popular book and turn it into an app where we could add different features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes about the book, etc. On top of that, we could also embed a dictionary within the app, so users don't need to exit our app to look up words in an external app.

This idea seems to fit well with the fact that the App Store is dominated by for-fun apps. This suggests the market might be a bit saturated with for-fun apps, which means a practical app might have more of a chance to stand out among the huge number of apps on the App Store.

Other genres that seem popular include weather, book, food and drink, or finance. The book genre seem to overlap a bit with the app idea we described above, but the other genres don't seem too interesting to us:

1. Weather apps — people generally don't spend too much time in-app, and the chances of making profit from in-app adds are low. Also, getting reliable live weather data may require us to connect our apps to non-free APIs.

2. Food and drink — examples here include Starbucks, Dunkin' Donuts, McDonald's, etc. So making a popular food and drink app requires actual cooking and a delivery service, which is outside the scope of our company.

3. Finance apps — these apps involve banking, paying bills, money transfer, etc. Building a finance app requires domain knowledge, and we don't want to hire a finance expert just to build an app.

Now let's analyze the Google Play market a bit.

# Most Popular Apps by Genre on Google Play
For the Google Play market, we actually have data about the number of installs, so we should be able to get a clearer picture about genre popularity. However, the install numbers don't seem precise enough — we can see that most values are open-ended (100+, 1,000+, 5,000+, etc.):

In [181]:
android_final.head()

App        Category  Rating  \
0  Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
3                           Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4           Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   
5                      Paper flowers instructions  ART_AND_DESIGN     4.4   
6         Smoke Effect Photo Maker - Smoke Editor  ART_AND_DESIGN     3.8   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   
5     167  5.6M      50,000+  Free     0       Everyone   
6     178   19M      50,000+  Free     0       Everyone   

                    Genres     Last Updated         Current Ver   Android Ver  
0             Art & Design  January 7, 2018               1.0.0  4.0.3 and up  
3             Art & Design     June 8, 2018  Varies with device    4.2 and up  
4  Art & Design;Creativity    June 20, 2018                 1.1    4.4 and up  
5             Art & Design   March 26, 2017                 1.0    2.3 and up  
6             Art & Design   April 26, 2018                 1.1  4.0.3 and up

In [191]:
(android_final.groupby('Installs')['Genres'].count() / (android_final.groupby('Installs')['Genres'].count()).sum()) * 100

Installs
0                  0.011896
0+                 0.047585
1+                 0.511539
1,000+             8.482037
1,000,000+        15.607899
1,000,000,000+     0.226029
10+                3.545087
10,000+           10.444920
10,000,000+       10.290269
100+               7.090174
100,000+          11.598858
100,000,000+       2.117535
5+                 0.820842
5,000+             4.532477
5,000,000+         6.673804
50+                1.974780
50,000+            4.722817
50,000,000+        2.200809
500+               3.247680
500,000+           5.567452
500,000,000+       0.285510
Name: Genres, dtype: float64

In [188]:
(android_final.groupby('Installs')['Genres'].count()).sum()

8406

One problem with this data is that is not precise. For instance, we don't know whether an app with 100,000+ installs has 100,000 installs, 200,000, or 350,000. However, we don't need very precise data for our purposes — we only want to get an idea which app genres attract the most users, and we don't need perfect precision with respect to the number of users.

We're going to leave the numbers as they are, which means that we'll consider that an app with 100,000+ installs has 100,000 installs, and an app with 1,000,000+ installs has 1,000,000 installs, and so on.

To perform computations, however, we'll need to convert each install number to float — this means that we need to remove the commas and the plus characters, otherwise the conversion will fail and raise an error. We'll do this directly in the loop below, where we also compute the average number of installs for each genre (category).

In [192]:
android_final['Installs'] = android_final['Installs'].str.replace('+','')

C:\Users\MATHEW~1.OLA\AppData\Local\Temp/ipykernel_17500/1830992640.py:1: FutureWarning: The default value of regex will change from True to False in a future version. In addition, single character regular expressions will *not* be treated as literal strings when regex=True.
  android_final['Installs'] = android_final['Installs'].str.replace('+','')
C:\Users\MATHEW~1.OLA\AppData\Local\Temp/ipykernel_17500/1830992640.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  android_final['Installs'] = android_final['Installs'].str.replace('+','')


In [215]:
(android_final.groupby('Installs')['Genres'].count() / (android_final.groupby('Installs')['Genres'].count()).sum()) * 100

Installs
0.000000e+00     0.059481
1.000000e+00     0.511539
5.000000e+00     0.820842
1.000000e+01     3.545087
5.000000e+01     1.974780
1.000000e+02     7.090174
5.000000e+02     3.247680
1.000000e+03     8.482037
5.000000e+03     4.532477
1.000000e+04    10.444920
5.000000e+04     4.722817
1.000000e+05    11.598858
5.000000e+05     5.567452
1.000000e+06    15.607899
5.000000e+06     6.673804
1.000000e+07    10.290269
5.000000e+07     2.200809
1.000000e+08     2.117535
5.000000e+08     0.285510
1.000000e+09     0.226029
Name: Genres, dtype: float64

In [195]:
android_final['Installs'] = android_final['Installs'].str.replace(',','')

C:\Users\MATHEW~1.OLA\AppData\Local\Temp/ipykernel_17500/4015269984.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  android_final['Installs'] = android_final['Installs'].str.replace(',','')


In [196]:
android_final["Installs"] = pd.to_numeric(android_final["Installs"], downcast="float")

C:\Users\MATHEW~1.OLA\AppData\Local\Temp/ipykernel_17500/1729205233.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  android_final["Installs"] = pd.to_numeric(android_final["Installs"], downcast="float")


In [198]:
android_final.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 8406 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             8406 non-null   object 
 1   Category        8406 non-null   object 
 2   Rating          7175 non-null   float64
 3   Reviews         8406 non-null   object 
 4   Size            8406 non-null   object 
 5   Installs        8406 non-null   float32
 6   Type            8405 non-null   object 
 7   Price           8406 non-null   object 
 8   Content Rating  8406 non-null   object 
 9   Genres          8406 non-null   object 
 10  Last Updated    8406 non-null   object 
 11  Current Ver     8400 non-null   object 
 12  Android Ver     8405 non-null   object 
dtypes: float32(1), float64(1), object(11)
memory usage: 886.6+ KB


In [218]:
android_final[android_final['Installs'] == 1000000000]

App             Category  Rating   Reviews  \
152               Google Play Books  BOOKS_AND_REFERENCE     3.9   1433233   
381              WhatsApp Messenger        COMMUNICATION     4.4  69119316   
411    Google Chrome: Fast & Secure        COMMUNICATION     4.3   9643041   
451                           Gmail        COMMUNICATION     4.3   4604483   
464                        Hangouts        COMMUNICATION     4.0   3419513   
1917                 Subway Surfers                 GAME     4.5  27725352   
2544                       Facebook               SOCIAL     4.1  78158306   
2554                        Google+               SOCIAL     4.2   4831125   
2604                      Instagram               SOCIAL     4.5  66577446   
2884                  Google Photos          PHOTOGRAPHY     4.5  10859051   
3223      Maps - Navigate & Explore     TRAVEL_AND_LOCAL     4.3   9235373   
3232             Google Street View     TRAVEL_AND_LOCAL     4.2   2129707   
3234                         Google                TOOLS     4.4   8033493   
3523                   Google Drive         PRODUCTIVITY     4.4   2731211   
3665                        YouTube        VIDEO_PLAYERS     4.3  25655305   
3687        Google Play Movies & TV        VIDEO_PLAYERS     3.7    906384   
4234  Skype - free IM & video calls        COMMUNICATION     4.1  10484169   
5856              Google Play Games               FAMILY     4.3   7168735   
9844                    Google News   NEWS_AND_MAGAZINES     3.9    878065   

                    Size      Installs  Type Price Content Rating  \
152   Varies with device  1.000000e+09  Free     0           Teen   
381   Varies with device  1.000000e+09  Free     0       Everyone   
411   Varies with device  1.000000e+09  Free     0       Everyone   
451   Varies with device  1.000000e+09  Free     0       Everyone   
464   Varies with device  1.000000e+09  Free     0       Everyone   
1917                 76M  1.000000e+09  Free     0   Everyone 10+   
2544  Varies with device  1.000000e+09  Free     0           Teen   
2554  Varies with device  1.000000e+09  Free     0           Teen   
2604  Varies with device  1.000000e+09  Free     0           Teen   
2884  Varies with device  1.000000e+09  Free     0       Everyone   
3223  Varies with device  1.000000e+09  Free     0       Everyone   
3232  Varies with device  1.000000e+09  Free     0       Everyone   
3234  Varies with device  1.000000e+09  Free     0       Everyone   
3523  Varies with device  1.000000e+09  Free     0       Everyone   
3665  Varies with device  1.000000e+09  Free     0           Teen   
3687  Varies with device  1.000000e+09  Free     0           Teen   
4234  Varies with device  1.000000e+09  Free     0       Everyone   
5856  Varies with device  1.000000e+09  Free     0           Teen   
9844                 13M  1.000000e+09  Free     0           Teen   

                       Genres    Last Updated         Current Ver  \
152         Books & Reference  August 3, 2018  Varies with device   
381             Communication  August 3, 2018  Varies with device   
411             Communication  August 1, 2018  Varies with device   
451             Communication  August 2, 2018  Varies with device   
464             Communication   July 21, 2018  Varies with device   
1917                   Arcade   July 12, 2018              1.90.0   
2544                   Social  August 3, 2018  Varies with device   
2554                   Social   July 26, 2018  Varies with device   
2604                   Social   July 31, 2018  Varies with device   
2884              Photography  August 6, 2018  Varies with device   
3223           Travel & Local   July 31, 2018  Varies with device   
3232           Travel & Local  August 6, 2018  Varies with device   
3234                    Tools  August 3, 2018  Varies with device   
3523             Productivity  August 6, 2018  Varies with device   
3665  Video Players & Editors  August 2, 2018  Varies w

In [206]:
total1 = android_final.groupby('Category')['Installs'].sum()

In [207]:
total1

Category
ART_AND_DESIGN         1.082211e+08
AUTO_AND_VEHICLES      5.098006e+07
BEAUTY                 2.719705e+07
BOOKS_AND_REFERENCE    1.564873e+09
BUSINESS               6.347715e+08
COMICS                 4.226115e+07
COMMUNICATION          9.784906e+09
DATING                 1.178038e+08
EDUCATION              1.813000e+08
ENTERTAINMENT          9.703600e+08
EVENTS                 1.397315e+07
FAMILY                 5.754991e+09
FINANCE                4.233426e+08
FOOD_AND_DRINK         1.994687e+08
GAME                   1.240135e+10
HEALTH_AND_FITNESS     1.121338e+09
HOUSE_AND_HOME         9.460236e+07
LIBRARIES_AND_DEMO     5.129371e+07
LIFESTYLE              4.497222e+08
MAPS_AND_NAVIGATION    4.907053e+08
MEDICAL                3.638034e+07
NEWS_AND_MAGAZINES     2.351483e+09
PARENTING              2.996101e+07
PERSONALIZATION        1.397508e+09
PHOTOGRAPHY            4.570119e+09
PRODUCTIVITY           5.668814e+09
SHOPPING               1.381179e+09
SOCIAL             

In [210]:
total1 = total1.to_frame()

In [211]:
dfd1 = total1.div(tab1.values)
dfd1.columns = total1.columns + '/' + tab1.columns

In [212]:
dfd1

Installs/count
Category                           
ART_AND_DESIGN         1.932520e+06
AUTO_AND_VEHICLES      6.453172e+05
BEAUTY                 5.131519e+05
BOOKS_AND_REFERENCE    8.504746e+06
BUSINESS               1.602958e+06
COMICS                 8.804407e+05
COMMUNICATION          3.610666e+07
DATING                 7.649595e+05
EDUCATION              1.831313e+06
ENTERTAINMENT          1.244051e+07
EVENTS                 2.328858e+05
FAMILY                 3.633201e+06
FINANCE                1.348225e+06
FOOD_AND_DRINK         1.974937e+06
GAME                   1.540540e+07
HEALTH_AND_FITNESS     4.263642e+06
HOUSE_AND_HOME         1.391211e+06
LIBRARIES_AND_DEMO     6.749173e+05
LIFESTYLE              1.375297e+06
MAPS_AND_NAVIGATION    4.304432e+06
MEDICAL                1.192798e+05
NEWS_AND_MAGAZINES     1.000631e+07
PARENTING              5.447456e+05
PERSONALIZATION        5.027007e+06
PHOTOGRAPHY            1.806371e+07
PRODUCTIVITY           1.697250e+07
SHOPPING               7.307823e+06
SOCIAL                 2.444109e+07
SPORTS                 3.647640e+06
TOOLS                  1.096084e+07
TRAVEL_AND_LOCAL       1.448754e+07
VIDEO_PLAYERS          2.523461e+07
WEATHER                5.219217e+06

In [213]:
dfd1.sort_values('Installs/count',ascending =False)

Installs/count
Category                           
COMMUNICATION          3.610666e+07
VIDEO_PLAYERS          2.523461e+07
SOCIAL                 2.444109e+07
PHOTOGRAPHY            1.806371e+07
PRODUCTIVITY           1.697250e+07
GAME                   1.540540e+07
TRAVEL_AND_LOCAL       1.448754e+07
ENTERTAINMENT          1.244051e+07
TOOLS                  1.096084e+07
NEWS_AND_MAGAZINES     1.000631e+07
BOOKS_AND_REFERENCE    8.504746e+06
SHOPPING               7.307823e+06
WEATHER                5.219217e+06
PERSONALIZATION        5.027007e+06
MAPS_AND_NAVIGATION    4.304432e+06
HEALTH_AND_FITNESS     4.263642e+06
SPORTS                 3.647640e+06
FAMILY                 3.633201e+06
FOOD_AND_DRINK         1.974937e+06
ART_AND_DESIGN         1.932520e+06
EDUCATION              1.831313e+06
BUSINESS               1.602958e+06
HOUSE_AND_HOME         1.391211e+06
LIFESTYLE              1.375297e+06
FINANCE                1.348225e+06
COMICS                 8.804407e+05
DATING                 7.649595e+05
LIBRARIES_AND_DEMO     6.749173e+05
AUTO_AND_VEHICLES      6.453172e+05
PARENTING              5.447456e+05
BEAUTY                 5.131519e+05
EVENTS                 2.328858e+05
MEDICAL                1.192798e+05

On average, communication apps have the most installs: 36,106,660. This number is heavily skewed up by a few apps that have over one billion installs (WhatsApp, Facebook Messenger, Skype, Google Chrome, Gmail, and Hangouts), and a few others with over 100 and 500 million installs:

In [214]:
android_final.loc[android_final['Category'] == 'COMMUNICATION']

App       Category  Rating Reviews  \
337                     Messenger for SMS  COMMUNICATION     4.3  125257   
343                              My Tele2  COMMUNICATION     4.3  158679   
346          imo beta free calls and text  COMMUNICATION     4.3  659395   
348                              Contacts  COMMUNICATION     4.3   66602   
350                Web Browser & Explorer  COMMUNICATION     4.0   36901   
...                                   ...            ...     ...     ...   
10720  Firefox Focus: The privacy browser  COMMUNICATION     4.4   36981   
10734                          FP Connect  COMMUNICATION     NaN       0   
10739      FreedomPop Messaging Phone/SIM  COMMUNICATION     3.6    9894   
10748                             FP Live  COMMUNICATION     NaN       0   
10790              HipChat - beta version  COMMUNICATION     4.1    1035   

                     Size     Installs  Type Price Content Rating  \
337                   17M   10000000.0  Free     0           Teen   
343                  8.8M    5000000.0  Free     0       Everyone   
346                   11M  100000000.0  Free     0       Everyone   
348    Varies with device   50000000.0  Free     0       Everyone   
350                  6.6M    5000000.0  Free     0       Everyone   
...                   ...          ...   ...   ...            ...   
10720                4.0M    1000000.0  Free     0       Everyone   
10734                 22M        100.0  Free     0           Teen   
10739                 39M     500000.0  Free     0       Everyone   
10748                3.3M         10.0  Free     0           Teen   
10790                 20M      50000.0  Free     0       Everyone   

              Genres       Last Updated       Current Ver    Android Ver  
337    Communication       June 6, 2018             1.8.9     4.1 and up  
343    Communication     August 3, 2018             2.4.1     4.4 and up  
346    Communication       June 7, 2018  9.8.000000010492     4.0 and up  
348    Communication      June 26, 2018   2.8.4.201036949     5.0 and up  
350    Communication       July 4, 2018            11.8.6  4.0.3 - 7.1.1  
...              ...                ...               ...            ...  
10720  Communication       July 6, 2018               5.2     5.0 and up  
10734  Communication  December 15, 2017            3.15.1     4.1 and up  
10739  Communication      July 26, 2018   23.01.1265.0712     4.1 and up  
10748  Communication   November 3, 2017             1.2.4     4.2 and up  
10790  Communication     August 7, 2018          3.20.001     4.1 and up  

[271 rows x 13 columns]

In [236]:
android_final[(android_final['Installs'].isin([1000000000, 500000000, 100000000])) & (android_final['Category'] =='COMMUNICATION')]

App       Category  \
346                        imo beta free calls and text  COMMUNICATION   
381                                  WhatsApp Messenger  COMMUNICATION   
393                                                 Who  COMMUNICATION   
395          GO SMS Pro - Messenger, Free Themes, Emoji  COMMUNICATION   
396                                    Android Messages  COMMUNICATION   
411                        Google Chrome: Fast & Secure  COMMUNICATION   
412                      Firefox Browser fast & private  COMMUNICATION   
420         UC Browser - Fast Download Private & Secure  COMMUNICATION   
451                                               Gmail  COMMUNICATION   
464                                            Hangouts  COMMUNICATION   
465                       imo free video calls and chat  COMMUNICATION   
474                         LINE: Free Calls & Messages  COMMUNICATION   
4039              Google Duo - High Quality Video Calls  COMMUNICATION   
4106              Messenger Lite: Free Calls & Messages  COMMUNICATION   
4234                      Skype - free IM & video calls  COMMUNICATION   
4279                                                Kik  COMMUNICATION   
4302                       KakaoTalk: Free Calls & Text  COMMUNICATION   
4421                      Opera Mini - fast web browser  COMMUNICATION   
4422                     Opera Browser: Fast and Secure  COMMUNICATION   
4592                                           Telegram  COMMUNICATION   
4602  Truecaller: Caller ID, SMS spam blocking & Dialer  COMMUNICATION   
4633        UC Browser Mini -Tiny Fast Private & Secure  COMMUNICATION   
4676                                    Viber Messenger  COMMUNICATION   
4712                                             WeChat  COMMUNICATION   
6449                        BBM - Free Calls & Messages  COMMUNICATION   

      Rating   Reviews                Size      Installs  Type Price  \
346      4.3    659395                 11M  1.000000e+08  Free     0   
381      4.4  69119316  Varies with device  1.000000e+09  Free     0   
393      4.3   2451093  Varies with device  1.000000e+08  Free     0   
395      4.4   2876500                 24M  1.000000e+08  Free     0   
396      4.2    781810  Varies with device  1.000000e+08  Free     0   
411      4.3   9643041  Varies with device  1.000000e+09  Free     0   
412      4.4   3075118  Varies with device  1.000000e+08  Free     0   
420      4.5  17714850                 40M  5.000000e+08  Free     0   
451      4.3   4604483  Varies with device  1.000000e+09  Free     0   
464      4.0   3419513  Varies with device  1.000000e+09  Free     0   
465      4.3   4785988                 11M  5.000000e+08  Free     0   
474      4.2  10790289  Varies with device  5.000000e+08  Free     0   
4039     4.6   2083237  Varies with device  5.000000e+08  Free     0   
4106     4.4   1429038  Varies with device  1.000000e+08  Free     0   
4234     4.1  10484169  Varies with device  1.000000e+09  Free     0   
4279     4.3   2451136  Varies with device  1.000000e+08  Free     0   
4302     4.3   2546549  Varies with device  1.000000e+08  Free     0   
4421     4.5   5150801  Varies with device  1.000000e+08  Free     0   
4422     4.4   2473795  Varies with device  1.000000e+08  Free     0   
4592     4.4   3128611  Varies with device  1.000000e+08  Free     0   
4602     4.5   7820775  Varies with device  1.000000e+08  Free     0   
4633     4.4   3648765                3.3M  1.000000e+08  Free     0   
4676     4.3  11335481  Varies with device  5.000000e+08  Free     0   
4712     4.2   5387631  Varies with device  1.000000e+08  Free     0   
6449     4.3  12843436  Varies with device  1.000000e+08  Free     0   

     Content Rating         Genres    Last Updated               Current Ver  \
346        Everyone  Communication    June 7, 2018          9.8.000000010492   
381        Everyone  Communication  August 3, 2018        Varies with device   
393           

If we removed all the communication apps that have over 100 million installs, the average would be reduced roughly ten times:

In [237]:
andr = android_final.copy()

In [238]:
andr.head()

App        Category  Rating  \
0  Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
3                           Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4           Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   
5                      Paper flowers instructions  ART_AND_DESIGN     4.4   
6         Smoke Effect Photo Maker - Smoke Editor  ART_AND_DESIGN     3.8   

  Reviews  Size    Installs  Type Price Content Rating  \
0     159   19M     10000.0  Free     0       Everyone   
3  215644   25M  50000000.0  Free     0           Teen   
4     967  2.8M    100000.0  Free     0       Everyone   
5     167  5.6M     50000.0  Free     0       Everyone   
6     178   19M     50000.0  Free     0       Everyone   

                    Genres     Last Updated         Current Ver   Android Ver  
0             Art & Design  January 7, 2018               1.0.0  4.0.3 and up  
3             Art & Design     June 8, 2018  Varies with device    4.2 and up  
4  Art & Design;Creativity    June 20, 2018                 1.1    4.4 and up  
5             Art & Design   March 26, 2017                 1.0    2.3 and up  
6             Art & Design   April 26, 2018                 1.1  4.0.3 and up

In [240]:
andr.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 8406 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             8406 non-null   object 
 1   Category        8406 non-null   object 
 2   Rating          7175 non-null   float64
 3   Reviews         8406 non-null   object 
 4   Size            8406 non-null   object 
 5   Installs        8406 non-null   float32
 6   Type            8405 non-null   object 
 7   Price           8406 non-null   object 
 8   Content Rating  8406 non-null   object 
 9   Genres          8406 non-null   object 
 10  Last Updated    8406 non-null   object 
 11  Current Ver     8400 non-null   object 
 12  Android Ver     8405 non-null   object 
dtypes: float32(1), float64(1), object(11)
memory usage: 886.6+ KB


In [239]:
index_names = andr[ (andr['Category'] == 'COMMUNICATION') & (andr['Installs'] >= 100000000)].index

In [241]:
andr.drop(index_names, inplace = True)

In [242]:
andr.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 8381 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             8381 non-null   object 
 1   Category        8381 non-null   object 
 2   Rating          7150 non-null   float64
 3   Reviews         8381 non-null   object 
 4   Size            8381 non-null   object 
 5   Installs        8381 non-null   float32
 6   Type            8380 non-null   object 
 7   Price           8381 non-null   object 
 8   Content Rating  8381 non-null   object 
 9   Genres          8381 non-null   object 
 10  Last Updated    8381 non-null   object 
 11  Current Ver     8375 non-null   object 
 12  Android Ver     8380 non-null   object 
dtypes: float32(1), float64(1), object(11)
memory usage: 883.9+ KB


In [244]:
total2 = andr.groupby('Category')['Installs'].sum()

In [245]:
total2 = total2.to_frame()

In [246]:
dfd2 = total2.div(tab1.values)
dfd2.columns = total2.columns + '/' + tab1.columns

In [247]:
dfd2.sort_values('Installs/count',ascending =False)

Installs/count
Category                           
VIDEO_PLAYERS          2.523461e+07
SOCIAL                 2.444109e+07
PHOTOGRAPHY            1.806371e+07
PRODUCTIVITY           1.697250e+07
GAME                   1.540540e+07
TRAVEL_AND_LOCAL       1.448754e+07
ENTERTAINMENT          1.244051e+07
TOOLS                  1.096084e+07
NEWS_AND_MAGAZINES     1.000631e+07
BOOKS_AND_REFERENCE    8.504746e+06
SHOPPING               7.307823e+06
WEATHER                5.219217e+06
PERSONALIZATION        5.027007e+06
MAPS_AND_NAVIGATION    4.304432e+06
HEALTH_AND_FITNESS     4.263642e+06
SPORTS                 3.647640e+06
FAMILY                 3.633201e+06
COMMUNICATION          2.896330e+06
FOOD_AND_DRINK         1.974937e+06
ART_AND_DESIGN         1.932520e+06
EDUCATION              1.831313e+06
BUSINESS               1.602958e+06
HOUSE_AND_HOME         1.391211e+06
LIFESTYLE              1.375297e+06
FINANCE                1.348225e+06
COMICS                 8.804407e+05
DATING                 7.649595e+05
LIBRARIES_AND_DEMO     6.749173e+05
AUTO_AND_VEHICLES      6.453172e+05
PARENTING              5.447456e+05
BEAUTY                 5.131519e+05
EVENTS                 2.328858e+05
MEDICAL                1.192798e+05

We see the same pattern for the video players category, which is the runner-up with 24,727,872 installs. The market is dominated by apps like Youtube, Google Play Movies & TV, or MX Player. The pattern is repeated for social apps (where we have giants like Facebook, Instagram, Google+, etc.), photography apps (Google Photos and other popular photo editors), or productivity apps (Microsoft Word, Dropbox, Google Calendar, Evernote, etc.).

Again, the main concern is that these app genres might seem more popular than they really are. Moreover, these niches seem to be dominated by a few giants who are hard to compete against.

The game genre seems pretty popular, but previously we found out this part of the market seems a bit saturated, so we'd like to come up with a different app recommendation if possible.

The books and reference genre looks fairly popular as well, with an average number of installs of 8,767,811. It's interesting to explore this in more depth, since we found this genre has some potential to work well on the App Store, and our aim is to recommend an app genre that shows potential for being profitable on both the App Store and Google Play.

Let's take a look at some of the apps from this genre and their number of installs:

In [249]:
android_final[['Category','Installs']] [android_final['Category'] == 'BOOKS_AND_REFERENCE']

Category    Installs
140    BOOKS_AND_REFERENCE     50000.0
141    BOOKS_AND_REFERENCE    100000.0
142    BOOKS_AND_REFERENCE  10000000.0
144    BOOKS_AND_REFERENCE  10000000.0
147    BOOKS_AND_REFERENCE    100000.0
...                    ...         ...
10815  BOOKS_AND_REFERENCE    500000.0
10819  BOOKS_AND_REFERENCE      5000.0
10829  BOOKS_AND_REFERENCE     10000.0
10833  BOOKS_AND_REFERENCE      1000.0
10839  BOOKS_AND_REFERENCE      1000.0

[184 rows x 2 columns]

In [251]:
(android_final[['Category','Installs']] [android_final['Category'] == 'BOOKS_AND_REFERENCE']).describe()

Installs
count  1.840000e+02
mean   8.504744e+06
std    7.461520e+07
min    5.000000e+00
25%    1.000000e+03
50%    5.000000e+04
75%    1.000000e+06
max    1.000000e+09

The book and reference genre includes a variety of apps: software for processing and reading ebooks, various collections of libraries, dictionaries, tutorials on programming or languages, etc. It seems there's still a small number of extremely popular apps that skew the average:

In [252]:
android_final[(android_final['Installs'].isin([1000000000, 500000000, 100000000])) & (android_final['Category'] =='BOOKS_AND_REFERENCE')]

App             Category  Rating  Reviews  \
152         Google Play Books  BOOKS_AND_REFERENCE     3.9  1433233   
3941                    Bible  BOOKS_AND_REFERENCE     4.7  2440695   
4083            Amazon Kindle  BOOKS_AND_REFERENCE     4.2   814151   
5651  Audiobooks from Audible  BOOKS_AND_REFERENCE     4.5   568922   

                    Size      Installs  Type Price Content Rating  \
152   Varies with device  1.000000e+09  Free     0           Teen   
3941  Varies with device  1.000000e+08  Free     0           Teen   
4083  Varies with device  1.000000e+08  Free     0           Teen   
5651  Varies with device  1.000000e+08  Free     0           Teen   

                 Genres    Last Updated         Current Ver  \
152   Books & Reference  August 3, 2018  Varies with device   
3941  Books & Reference  August 2, 2018  Varies with device   
4083  Books & Reference   July 27, 2018  Varies with device   
5651  Books & Reference  August 1, 2018  Varies with device   

             Android Ver  
152   Varies with device  
3941  Varies with device  
4083  Varies with device  
5651  Varies with device

However, it looks like there are only a few very popular apps, so this market still shows potential. Let's try to get some app ideas based on the kind of apps that are somewhere in the middle in terms of popularity (between 1,000,000 and 100,000,000 downloads):

In [253]:
android_final[(android_final['Installs'].isin([1000000, 5000000, 10000000])) & (android_final['Category'] =='BOOKS_AND_REFERENCE')]

App             Category  \
142                                           Wikipedia  BOOKS_AND_REFERENCE   
144                                         Cool Reader  BOOKS_AND_REFERENCE   
148                                          Book store  BOOKS_AND_REFERENCE   
149                      FBReader: Favorite Book Reader  BOOKS_AND_REFERENCE   
151          Free Books - Spirit Fanfiction and Stories  BOOKS_AND_REFERENCE   
153                      AlReader -any text book reader  BOOKS_AND_REFERENCE   
158                                   FamilySearch Tree  BOOKS_AND_REFERENCE   
159                                      Cloud of Books  BOOKS_AND_REFERENCE   
166                                   Read books online  BOOKS_AND_REFERENCE   
168                     eBoox: book reader fb2 epub zip  BOOKS_AND_REFERENCE   
171                                  All Maths Formulas  BOOKS_AND_REFERENCE   
172                                            Ancestry  BOOKS_AND_REFERENCE   
173                                            HTC Help  BOOKS_AND_REFERENCE   
179                                        Moon+ Reader  BOOKS_AND_REFERENCE   
181                          English-Myanmar Dictionary  BOOKS_AND_REFERENCE   
182                           Golden Dictionary (EN-AR)  BOOKS_AND_REFERENCE   
183                        All Language Translator Free  BOOKS_AND_REFERENCE   
4095                                       Ebook Reader  BOOKS_AND_REFERENCE   
4100                                 Aldiko Book Reader  BOOKS_AND_REFERENCE   
4740                               Dictionary - WordWeb  BOOKS_AND_REFERENCE   
5076                50000 Free eBooks & Free AudioBooks  BOOKS_AND_REFERENCE   
5319                                    Al-Quran (Free)  BOOKS_AND_REFERENCE   
5323                                 Al Quran Indonesia  BOOKS_AND_REFERENCE   
5324                          Al'Quran Bahasa Indonesia  BOOKS_AND_REFERENCE   
5325                                  Al Quran Al karim  BOOKS_AND_REFERENCE   
5334      Al Quran : EAlim - Translations & MP3 Offline  BOOKS_AND_REFERENCE   
5339                     Koran Read &MP3 30 Juz Offline  BOOKS_AND_REFERENCE   
5342                     Hafizi Quran 15 lines per page  BOOKS_AND_REFERENCE   
5345                                  Quran for Android  BOOKS_AND_REFERENCE   
5561                                       Satellite AR  BOOKS_AND_REFERENCE   
5899                        Oxford A-Z of English Usage  BOOKS_AND_REFERENCE   
6290  Dictionary.com: Find Definitions for English W...  BOOKS_AND_REFERENCE   
6296                       English Dictionary - Offline  BOOKS_AND_REFERENCE   
6298                                          Bible KJV  BOOKS_AND_REFERENCE   
6497                      NOOK: Read eBooks & Magazines  BOOKS_AND_REFERENCE   
6630  Brilliant Quotes: Life, Love, Family & Motivation  BOOKS_AND_REFERENCE   
7780                      Stats Royale for Clash Royale  BOOKS_AND_REFERENCE   
8293                                         Dictionary  BOOKS_AND_REFERENCE   
8644                        wikiHow: how to do anything  BOOKS_AND_REFERENCE   
9338                                       EGW Writings  BOOKS_AND_REFERENCE   
9344                            My Little Pony AR Guide  BOOKS_AND_REFERENCE   
9570                         Spanish English Translator  BOOKS_AND_REFERENCE   
9621                       Dictionary - Merriam-Webster  BOOKS_AND_REFERENCE   
9625                                         JW Library  BOOKS_AND_REFERENCE   
9642                Oxford Dictionary of English : Free  BOOKS_AND_REFERENCE   
9643                           English Hindi Dictionary  BOOKS_AND_REFERENCE   
9644                        English to Hindi Dictionary  BOOKS_AND_REFERENCE   

      Rating Reviews                Size    Installs  Type Price  \
142      4.4  577550  Varies with device  10000000.0  Free     0   
144      4.5  246315  Varies with device  10000000.0  Free     0   
148    

This niche seems to be dominated by software for processing and reading ebooks, as well as various collections of libraries and dictionaries, so it's probably not a good idea to build similar apps since there'll be some significant competition.

We also notice there are quite a few apps built around the book Quran, which suggests that building an app around a popular book can be profitable. It seems that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets.

However, it looks like the market is already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.

# Conclusions
In this project, we analyzed data about the App Store and Google Play mobile apps with the goal of recommending an app profile that can be profitable for both markets.

We concluded that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets. The markets are already full of libraries, so we need to add some special features besides the raw version of the book. This might include daily quotes from the book, an audio version of the book, quizzes on the book, a forum where people can discuss the book, etc.